sympy/geometry/ellipse.py

```diff
@@ -1339,28 +1341,78 @@ class Circle(Ellipse):
     >>> c2.hradius, c2.vradius, c2.radius, c2.center
     (sqrt(2)/2, sqrt(2)/2, sqrt(2)/2, Point2D(1/2, 1/2))
 
+    >>> # a circle object is returned
+    >>> from sympy.abc import x, y, a, b
+    >>> Circle(x ** 2 + y ** 2 - 25)
+    Circle(Point2D(0, 0), 5)
+    >>> Circle(a ** 2 + b ** 2 - 25, x='a', y='b')
+    Circle(Point2D(0, 0), 5)
+
     """
 
     def __new__(cls, *args, **kwargs):
-        c, r = None, None
-        if len(args) == 3:
-            args = [Point(a, dim=2) for a in args]
-            if Point.is_collinear(*args):
-                raise GeometryError(
-                    "Cannot construct a circle from three collinear points")
-            from .polygon import Triangle
-            t = Triangle(*args)
-            c = t.circumcenter
-            r = t.circumradius
-        elif len(args) == 2:
-            # Assume (center, radius) pair
-            c = Point(args[0], dim=2)
-            r = sympify(args[1])
-
-        if not (c is None or r is None):
-            return GeometryEntity.__new__(cls, c, r, **kwargs)
-
-        raise GeometryError("Circle.__new__ received unknown arguments")
+
+        if len(args) == 1 and isinstance(args[0], Expr):
+            x = kwargs.get('x', 'x')
+            y = kwargs.get('y', 'y')
+            equation = args[0]
+
+            def find(x_, equation_):
+                free = equation_.free_symbols
+                xs = [i for i in free if (i.name if type(x_) is str else i) == x_]
+                if not xs:
+                    raise ValueError('could not find %s' % x_)
+                if len(xs) != 1:
+                    raise ValueError('ambiguous %s' % x_)
+                return xs[0]
+
+            x = find(x, equation)
+            y = find(y, equation)
+            poly_x = poly(equation, x)
```

Comment:
This looks like it could be done more succinctly. You can create a Poly in x**2 and y**2 and then check to see that the domain doesn't contain x or y. Then queries regarding coefficients and the independent term should be easy.